Delta dereg details

rtorrero opened the pull request:
# Description

Add details views changes for delta dereg

arbulu89 commented on assets/js/components/SapSystemDetails/InstanceStatus.jsx:
```diff
@@ -1,31 +1,33 @@
   return (
-    <Pill roundedMode="rounded" className={`${cssClass} text-gray-50`}>
+    <Pill className="bg-gray-200 text-gray-500 items-center">
+      SAPControl: <EOS_LENS_FILLED size="base" className={`${cssClass} mx-1`} />{' '}
```
What is this final `{' '}` for? To add an empty space? There is not a better way of doing that?
You already have a space after the current `:` for example

arbulu89 commented on assets/js/components/SapSystemDetails/InstanceStatus.jsx:
```diff
@@ -1,31 +1,33 @@
 import React from 'react';
 import Pill from '@components/Pill/Pill';
+import { EOS_LENS_FILLED } from 'eos-icons-react';
```
I would put this `import` before importing `Pill`.
We put the 3rd party imports at the top usually, before our own component imports

arbulu89 commented on assets/js/components/SapSystemDetails/SapSystemDetails.stories.jsx:
```diff
@@ -48,3 +59,11 @@ export const SapSystem = {
+
+export const SapSystemWithAbsentInstance = {
+  args: {
+    title: 'SAP System Details',
```
We could have this as to avoid a bit of duplication. Not a big deal:
```
args: {
    ...SapSystem,
    system: systemWithAbsentInstance,
}
```

arbulu89 commented on assets/js/components/DatabaseDetails/DatabaseDetails.stories.jsx:
```diff
@@ -16,6 +16,13 @@ const database = {
+};
+databaseWithAbsentInstance.instances[1].absent_at = faker.date
+  .past()
+  .toISOString();
```
Let's add some new lines to split the lines and make it a bit more readable

arbulu89 commented on assets/js/components/DatabaseDetails/DatabaseDetails.stories.jsx:
```diff
@@ -46,3 +53,11 @@ export const Database = {
+
+export const DatabaseWithAbsentInstance = {
+  args: {
+    title: 'Database Details',
```
The same to reuse some data as in SAP system story

arbulu89 commented on assets/js/components/SapSystemDetails/GenericSystemDetails.test.jsx:
```diff
@@ -73,4 +73,27 @@ describe('GenericSystemDetails', () => {
+    expect(screen.queryByRole('button', { name: 'Clean up' })).toBeVisible();
+    const [_sapSystemIcon, health, _cleanUpIcon] =
+      screen.getAllByTestId('eos-svg-component');
+    expect(health).toHaveClass('fill-black');
```
We could even test the tooltip

arbulu89 commented on assets/js/components/SapSystemDetails/tableConfigs.jsx:
```diff
@@ -1,29 +1,101 @@
 
 export const systemInstancesTableConfiguration = {
   usePadding: false,
+  rowClassName: 'bg-jungle-green',
```
What is this for?

rtorrero replied:
Forgot to remove, this was a small test

arbulu89 commented on assets/js/components/SapSystemDetails/tableConfigs.jsx:
```diff
@@ -1,29 +1,101 @@
+              isEnabled={!!item.absent_at}
+            >
+              <HealthIcon health="absent" />
+              <span className="mx-1 text-gray-600">{content}</span>
```
Should this span go within the tooltip?
I mean, i expected the tooltip to be available only in the icon

arbulu89 commented on test/e2e/cypress/e2e/sap_system_details.cy.js:
```diff
@@ -60,23 +60,23 @@ context('SAP system details', () => {
         cy.get('@tableCell')
           .eq(6)
-          .find('span')
-          .should('have.class', healthMap[instance.StatusBadge]);
```
Can we remove this `StatusBadge` from the fixtures maybe?
If it is not used anymore...

arbulu89 commented on assets/js/components/SapSystemDetails/tableConfigs.jsx:
```diff
@@ -1,29 +1,101 @@
+      key: 'absent_at',
+      className: 'w-40',
+      render: (content, _item) => {
+        if (content) {
```
```
content && (
    <CleanUpButton
...
```
like in the `HostsList` component looks better

arbulu89 commented on assets/js/components/SapSystemDetails/tableConfigs.jsx:
```diff
@@ -1,29 +1,101 @@
+      title: 'Instance nr',
+      key: 'instance_number',
+      render: (content, item) => (
+        <span className={classNames({ 'text-gray-600': !!item.absent_at })}>
```
I'm not sure if there is any other way for calculating the class, but in any case, could we have some generic `render` function for this as it is repeated many times here?
```
const cellRender = (content, item) => (        <span className={classNames({ 'text-gray-600': !!item.absent_at })}>{content}</span>)```

arbulu89 reviewed (commented):
What would be missing here?
- The row gray color
- The dispatch of the action as we do in https://github.com/trento-project/web/pull/1776

Am I missing something?

rtorrero commented:
> What would be missing here?
> 
>     * The row gray color
> 
>     * The dispatch of the action as we do in [Dispatch deregistering actions on clean up flow #1776](https://github.com/trento-project/web/pull/1776)
> 
>     * Navigate to the overview page if the system is completely deregistered
> 
> 
> Am I missing something? Not saying that you need to include all these things in this PR (i would just include the 1st). Just summarazing what's missing

I've addressed the code comments, but regarding to the 3 points: regarding the first one, it was a intentional omission as I had a chat with @dottorblaster on slack and it seems right now that this wouldn't be possible with the current Table component. I don't want to speak for him, but he expressed concerns about our components becoming too flexible and about the cost this would have long-term. I'm adding him to the conversation in case he wants to further explain. Adding also @jagabomb as he was behind the design.

Regarding the other 2 points, I'll address them in separate PRs if that's ok?

dottorblaster commented:
We can address the row classname in a separate PR I think :+1:

arbulu89 commented on assets/js/components/SapSystemDetails/GenericSystemDetails.test.jsx:
```diff
@@ -73,4 +73,27 @@ describe('GenericSystemDetails', () => {
   });
+
+  it('should render a cleanup button and correct health icon when absent instances exist', () => {
+    const sapSystem = keysToCamel(
```
Why this `keysToCamel`? We don't serialize this way the redux state anymore
I guess you copied that from old tests, but I'm not even sure how they work.
I forgot to update those apparently...
We need to remove that function from the tests

arbulu89 commented on assets/js/components/SapSystemDetails/tableConfigs.jsx:
```diff
@@ -1,29 +1,84 @@
+      render: cellRender,
+    },
+    {
+      title: 'Start Prio',
```
For some reason you changed `Priority` by `Prio`. I typo I guess hehe

rtorrero replied:
I noticed that it was written like that on the Figma link within the task in Jira, I guess the logic here is that it's a common abbreviation and saves us some space.

arbulu89 commented on assets/js/components/SapSystemDetails/tableConfigs.jsx:
```diff
@@ -1,29 +1,84 @@
+      title: 'Hostname',
+      key: 'instance_hostname',
+      render: (content, item) => {
+        if (item.absent_at) {
```
Maybe:
```
<>
{item.absent_at && (
    <span className="mr-1">
        <Tooltip content="Registered instance not found." place="bottom">
            <HealthIcon health="absent" />
        </Tooltip>
    </span>})
{renderCell(content, item)}
</>
```
Almost the same anyway

arbulu89 reviewed (approved):
It looks good Ruben.
I dropped some final tiny comments.
You don't need a new review, but feel free to put me again if you want.
